### PR TITLE
Update of VCPKG

### PR DESCRIPTION
**Pull request by beckerhe**
This branch updates vcpkg to the current version from master (from several days ago).

## Immediate advantages
- Current version of vcpkg bootstraps also with VS2019 (there was a bug in the previous version)
- Several dependencies have better CMake support (especially curl, breakpad and several more)
- Several hacks are not necessary anymore (freetype-gl was fixed for example)

## Qt5
Qt5 can also be built now with vcpkg (there was a bug in the old version which prevented that). This reduces the need for a developer to install qt on its own way.

## Deployment (Windows)
Installing dependencies with vcpkg has the advantage that automatic deployment of shared libraries is done. So the build folder of an executable target is (should) be fully self-sustained. I checked with OrbitQt and it works in my case. All the libraries are copied next to the OrbitQt.exe. The only exception is the Visual Studio Runtime which should be installed independently anyway.

### CODE REVIEW

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -25,6 +25,17 @@ via the Windows Installer from cmake.org are also viable options.
 Check out the `bootstrap-orbit.{sh, bat}` file for a general description on how
 to install dependencies and on how to set up a `build/`-directory.
 
+## Dependencies
+
+All our 3rd party dependencies are either included via git submodules or they are
+managed by vcpkg. But there are several dependencies vcpkg needs to build our
+dependencies from source. This especially applies to Linux where it is expected
+to have several libraties on tools to be provided by the system. Check out
```
[the remark this answers is not in the document]
Comment (beckerhe) [reply DEVELOPMENT.md]: Well, it seems the 'r' was out of stock.
Thank you for telling me. :+1: 

Fixed.

**Comment by beckerhe** [comment]
> Hi Henning, thanks for the PR, this will simplify further the build process and fix a lot of issues!
> 
> On Linux however, it seems like we're missing a dependency on GLib, I'm running into these linking errors:
> 
> /usr/bin/ld: ../external/vcpkg/installed/x64-linux/lib/libQt5Core.a(qeventdispatcher_glib.o): in function `QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)': qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x63): undefined reference to `g_main_context_iteration'
> /usr/bin/ld: qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x7f): undefined reference to `g_main_context_iteration' /usr/bin/ld: qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0xf0): undefined reference to `g_main_context_iteration'
> /usr/bin/ld: qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x11e): undefined reference to `g_main_context_iteration'
> 
> Was this building on Linux on your end?

Yes. It builds on my machine. But I don't really know why. I checked the cause of this and it should also break on my machine. I tried to build yesterday with my private machine and it breaks there as well.

The problem was reported several times to Qt and to vcpkg, see:
https://github.com/microsoft/vcpkg/issues/9442
https://bugreports.qt.io/browse/QTBUG-38913
https://bugreports.qt.io/browse/QTBUG-81113

It seems to be fixed in the latest release of Qt (5.14) which is not yet available in vcpkg.
I have a work around for this, which should fix it for now. I propose we should update vcpkg
again when Qt 5.14 is packaged.

**Comment by beckerhe** [comment]
> > Hi Henning, thanks for the PR, this will simplify further the build process and fix a lot of issues!
> > On Linux however, it seems like we're missing a dependency on GLib, I'm running into these linking errors:
> > /usr/bin/ld: ../external/vcpkg/installed/x64-linux/lib/libQt5Core.a(qeventdispatcher_glib.o): in function `QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>)': qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x63): undefined reference to `g_main_context_iteration'
> > /usr/bin/ld: qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x7f): undefined reference to `g_main_context_iteration' /usr/bin/ld: qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0xf0): undefined reference to `g_main_context_iteration'
> > /usr/bin/ld: qeventdispatcher_glib.cpp:(.text._ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x11e): undefined reference to `g_main_context_iteration'
> > Was this building on Linux on your end?
> 
> Yes. It builds on my machine. But I don't really know why. I checked the cause of this and it should also break on my machine. I tried to build yesterday with my private machine and it breaks there as well.
> 
> The problem was reported several times to Qt and to vcpkg, see:
> [microsoft/vcpkg#9442](https://github.com/microsoft/vcpkg/issues/9442)
> https://bugreports.qt.io/browse/QTBUG-38913
> https://bugreports.qt.io/browse/QTBUG-81113
> 
> It seems to be fixed in the latest release of Qt (5.14) which is not yet available in vcpkg.
> I have a work around for this, which should fix it for now. I propose we should update vcpkg
> again when Qt 5.14 is packaged.

Status update: The workaround which worked on my private machine breaks the build on my workstation at work. I tried to build Orbit on my Chromebook (in this new containerized Linux) and it has the same behaviour as my workstation (works fine without workaround / workaround breaks build). So probably I have to wait until tonight to have my private machine available.

**Comment by beckerhe** [comment]
I believe I figured it out. Qt5 links to glib if glib-headers are available. So it's the same problem as with freeglut and libxrandr.

I just added a patch to explicitly disable glib in all cases, which has to be applied before installing(building) qt5 via vcpkg.

@pierricgimmig Can you check if bootstrap-orbit.sh works now for you?

Thanks,
Henning

**Comment by pierricgimmig** [comment]
> @pierricgimmig Can you check if bootstrap-orbit.sh works now for you?

Hi @beckerhe , thanks for looking into it.  With your latest changes, the linking errors are gone.  I do however run into another issue when trying to launch Orbit:

qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Did you get that on your side as well?

**Comment by beckerhe** [comment]
> > @pierricgimmig Can you check if bootstrap-orbit.sh works now for you?
> 
> Hi @beckerhe , thanks for looking into it. With your latest changes, the linking errors are gone. I do however run into another issue when trying to launch Orbit:
> 
> qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
> 
> Did you get that on your side as well?

Yes, I also get that. And I've been looking into that this afternoon but haven't solved it yet.
Somehow it's weird because I was pretty sure it was working on this branch before I made the pull request. Well, but maybe I had some unclean build directory or some environment variable pointing to my local Qt installation. Anyway, I will hopefully solve that tomorrow.

Thank you for trying and sorry for the hassle.

Cheers,
Henning

**Comment by beckerhe** [comment]
Hi @pierricgimmig,
it seems we are finally getting somewhere. I changed the qt5 dependency to dynamic linking and now it works. At least on my machine.

Furthermore, I hopefully fixed the problem @ronaldfw mentioned, about the missing shaders and missing font file. CMake now copies the shader files and the truetype file over to the build directory and the references to theses files (relative to the executable).

What's not in here: On Linux the built qt5 libraries are not automatically copied to the build directory since this is not necessary in a development setup (thanks to RPATH). For packaging this would be necessary. But since there are currently no plans on deploying Linux-UI-binaries I haven't put much effort into this.

**Comment by pierricgimmig** [comment]
> Hi @pierricgimmig,
> it seems we are finally getting somewhere. I changed the qt5 dependency to dynamic linking and now it works. At least on my machine.
> 
> Furthermore, I hopefully fixed the problem @ronaldfw mentioned, about the missing shaders and missing font file. CMake now copies the shader files and the truetype file over to the build directory and the references to theses files (relative to the executable).
> 
> What's not in here: On Linux the built qt5 libraries are not automatically copied to the build directory since this is not necessary in a development setup (thanks to RPATH). For packaging this would be necessary. But since there are currently no plans on deploying Linux-UI-binaries I haven't put much effort into this.

Hi @beckerhe , many thanks for fixing the remaining issues.  I built from scratch on gLinux and everything seems to work fine.  Concerning packaging on Linux, it will come up...   Anyhow, we'll cross that bridge when we can't there :-)  I'm merging the PR.

**Pull request merged**